File: domain/src/main/java/io/prizy/domain/contest/model/CreateContest.java
package io.prizy.domain.contest.model;

import java.time.Instant;
import java.util.Collection;
import java.util.Optional;
import java.util.UUID;

import lombok.Builder;

@Builder
public record CreateContest(
  String name,
  String description,
  ContestCategory category,
  Instant fromDate,
  Instant toDate,
  Collection<String> assetIds,
  String coverAssetId,
  Collection<CreatePack> packs,
  Integer cost,
  Optional<String> facebookPage,
  Optional<String> instagramPage,
  Optional<String> website,
  Boolean newsletterSubscription,
  Boolean adultOnly,
  Merchant merchant,
  UUID boardId
) {
}

File: domain/src/main/kotlin/io/prizy/domain/contest/command/CreateContest.kt
package io.prizy.domain.contest.command

import io.prizy.domain.contest.model.ContestCategory
import io.prizy.domain.contest.model.Merchant
import java.time.Instant
import java.util.UUID

data class CreateContest(
  val name: String,
  val description: String,
  val category: ContestCategory,
  val fromDate: Instant,
  val toDate: Instant,
  val mediaIds: Collection<String>,
  val coverMediaId: String,
  val packs: Collection<CreatePack>,
  val cost: Int,
  val facebookPage: String?,
  val instagramPage: String?,
  val website: String?,
  val newsletterSubscription: Boolean,
  val adultOnly: Boolean,
  val merchant: Merchant,
  val boardId: UUID
)
